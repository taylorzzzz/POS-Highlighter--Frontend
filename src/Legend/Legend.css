@import "../colors.css";

.Legend {
  padding: 0px;
  min-width: 125px;
  font-size: 12px;
}

.buttons {
  margin-top: 20px;
}

button {
  padding: 0;
  cursor: pointer;
  color: var(--color-text);
  background: none;
  border: none;
  display: block;
  margin: 10px 0;
  outline: none;
  font-size: 12px;

  &:hover {
    color: var(--color-text);
    transition: .2s all;
  }
}