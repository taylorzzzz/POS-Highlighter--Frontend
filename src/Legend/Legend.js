import React from 'react';

import LegendItem from './LegendItem/LegendItem';

import './Legend.css';

function Legend({ 
    items = {}, 
    toggleSelection, 
    deselectAll,
    selectAll,
    selectDefaults,
    toggleEdit,
    editActive,
    changeColor,
    handleClose }) {
  
  const legendItems = Object.keys(items).map((key, i) => {

    const el = items[key];

    return <LegendItem 
      pos={key}
      item={el}
      key={i} 
      toggleSelection={toggleSelection}
      editActive={editActive} 
      changeColor={ changeColor }
      handleClose={ handleClose }/>
  })


  let editValues = editActive
                    ? ['Close Edit', 'close']
                    : ['Edit', 'edit'];

  return (
    <div className="Legend">

      <div onClick={toggleEdit} className="edit-button button">
        <i className="material-icons">{editValues[1]}</i>
        <span>{editValues[0]}</span>
      </div>

      { legendItems }

      <div className="select-buttons">

        <div className="button select-all-btn" onClick={selectAll}>
          <i className="material-icons">select_all</i>
          <span>Select All</span>
        </div>

        <div className="button deselect-all-btn" onClick={deselectAll}>
          <i className="material-icons">clear_all</i>
          <span>Deselect All</span>
        </div>

        <div className="button select-defaults-btn" onClick={selectDefaults}>
          <i className="material-icons">autorenew</i>
          <span>Select Defaults:  
            <span className="noun"> N</span>
            <span className="verb"> V</span>
          </span>
        </div>
      
      </div>

    </div>
  )

}

export default Legend;