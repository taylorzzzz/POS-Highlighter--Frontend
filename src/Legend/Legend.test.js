import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Legend from './Legend';

test('Legend renders properly', () => {
    const tree = renderer.create(<Legend />).toJSON();
    
    expect(tree).toMatchSnapshot();
})

test('Legend selectAll events properly', () => {

    const selectSpy = jest.fn();

    const wrapper = shallow(<Legend selectAll={selectSpy}/>);

    expect(selectSpy).toHaveBeenCalledTimes(0);

    const btn = wrapper.find('.select-all-btn');
    btn.simulate('click');
    
    expect(selectSpy).toHaveBeenCalled();
})

test('Legend deselectAll events properly', () => {

    const deselectSpy = jest.fn();

    const wrapper = shallow(<Legend deselectAll={deselectSpy}/>);

    expect(deselectSpy).toHaveBeenCalledTimes(0);

    const btn = wrapper.find('.deselect-all-btn');
    btn.simulate('click');
    
    expect(deselectSpy).toHaveBeenCalled();
})

test('Legend select default events properly', () => {

    const selectDefaultSpy = jest.fn();

    const wrapper = shallow(<Legend selectDefaults={selectDefaultSpy}/>);

    expect(selectDefaultSpy).toHaveBeenCalledTimes(0);

    const btn = wrapper.find('.select-defaults-btn');
    btn.simulate('click');
    
    expect(selectDefaultSpy).toHaveBeenCalled();
})

test('Legend toggleEdit events properly', () => {

    const toggleSpy = jest.fn();
    
    const wrapper = shallow(
        <Legend toggleEdit={toggleSpy}/>);

    expect(toggleSpy).toHaveBeenCalledTimes(0);

    const btn = wrapper.find('.edit-button');
    btn.simulate('click');
    
    expect(toggleSpy).toHaveBeenCalled();

})