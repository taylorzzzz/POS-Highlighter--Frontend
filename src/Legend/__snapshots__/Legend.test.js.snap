// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Legend renders properly 1`] = `
<div
  className="Legend"
>
  <div
    className="edit-button button"
    onClick={undefined}
  >
    <i
      className="material-icons"
    >
      edit
    </i>
    <span>
      Edit
    </span>
  </div>
  <div
    className="select-buttons"
  >
    <div
      className="button select-all-btn"
      onClick={undefined}
    >
      <i
        className="material-icons"
      >
        select_all
      </i>
      <span>
        Select All
      </span>
    </div>
    <div
      className="button deselect-all-btn"
      onClick={undefined}
    >
      <i
        className="material-icons"
      >
        clear_all
      </i>
      <span>
        Deselect All
      </span>
    </div>
    <div
      className="button select-defaults-btn"
      onClick={undefined}
    >
      <i
        className="material-icons"
      >
        autorenew
      </i>
      <span>
        Select Defaults:
        <span
          className="noun"
        >
           N
        </span>
        <span
          className="verb"
        >
           V
        </span>
      </span>
    </div>
  </div>
</div>
`;
